compiler/rustc_borrowck/src/diagnostics/region_errors.rs

```diff
@@ -70,7 +72,43 @@ impl<'tcx> ConstraintDescription for ConstraintCategory<'tcx> {
 ///
 /// Usually we expect this to either be empty or contain a small number of items, so we can avoid
 /// allocation most of the time.
-pub(crate) type RegionErrors<'tcx> = Vec<RegionErrorKind<'tcx>>;
+///
+/// This is just a wrapper around `Vec<RegionErrorKind<'tcx>>`,
+/// which eagerly calls `bug` on an error, if `-Ztreat-err-as-bug` is enabled.
+pub(crate) struct RegionErrors<'tcx>(Vec<RegionErrorKind<'tcx>>, bool);
```

Comment:
cc @estebank for thoughts on this

It might be nicer to actually integrate the delaying of errors in rustc_borrowck, but it isn't possible atm.

Reply:
@estebank ping about this comment, I think it's still standing

Reply:
I see that this code is no longer in this PR. We have a few mechanisms for stashing diagnostics away for later emission.

> it isn't possible atm.

Why is so?

Reply:
Well, its been added in a different PR since I originally opened this, so not relevant anymore.